subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -133,6 +139,91 @@ img_mgmt_state_flags(int query_slot)
 }
 #endif
 
+/* Return number of slot that is considered candidate for next boot */
+int img_mgmt_get_opposite_slot(int slot)
+{
+	if (slot < 0 ||
+	    slot > (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER) << 1) {
```

Comment:
`>=`? If 2 images, slots are 0, 1, 2, 3 but 2 << 1 is 4

Reply:
Good catch.